daemon/images/image_prune.go

```diff
@@ -47,12 +47,9 @@ func (i *ImageService) ImagesPrune(ctx context.Context, pruneFilters filters.Arg
 	rep := &types.ImagesPruneReport{}
 
 	danglingOnly := true
-	if pruneFilters.Contains("dangling") {
-		if pruneFilters.ExactMatch("dangling", "false") || pruneFilters.ExactMatch("dangling", "0") {
-			danglingOnly = false
-		} else if !pruneFilters.ExactMatch("dangling", "true") && !pruneFilters.ExactMatch("dangling", "1") {
-			return nil, invalidFilter{"dangling", pruneFilters.Get("dangling")}
-		}
+	err = pruneFilters.ParseBoolIfPresent(&danglingOnly, "dangling")
+	if err != nil {
```

Comment:
Wondering if it would make for an easier "api" to make this something like `filters.GetBool("name", <default>)`


Thinking of which, we have a `httputils.BoolValueOrDefault` (although that's in the `api/server` package, so I haven't fully checked what the consequences for that are). I recall I was looking at some of the `httputils` package to make (some of) them return a proper error (which could be "ignored" by the caller where suitable).

I'm also not sure if that function follows the same semantics for "what's valid" (looks like we may be a bit inconsistent there)